### PR TITLE
[PLUTO-6995] Added OverlimitResSettingsRule resource

### PR DESCRIPTION
**Pull request by infkfwlrm**

### CODE REVIEW

File: wallarm/resource_rule_overlimit_res_settings.go [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"point": {
```
Comment (concentrator) [inline wallarm/resource_rule_overlimit_res_settings.go:178]: Do we really need point here?

File: wallarm/resource_rule_overlimit_res_settings.go [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+		if err != nil {
+			return nil, err
+		}
+		ruleID, err := strconv.Atoi(idAttr[2])
+		if err != nil {
+			return nil, err
+		}
+		d.Set("action_id", actionID)
+		d.Set("rule_id", ruleID)
+		d.Set("rule_type", "rate_limit")
```
Comment (concentrator) [inline wallarm/resource_rule_overlimit_res_settings.go]: Oops!

File: wallarm/resource_rule_overlimit_res_settings.go [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+
+		hint := &wallarm.HintRead{
+			Limit:     1000,
+			Offset:    0,
+			OrderBy:   "updated_at",
+			OrderDesc: true,
+			Filter: &wallarm.HintFilter{
+				Clientid: []int{clientID},
+				ID:       []int{ruleID},
+				Type:     []string{"rate_limit"},
```
Comment (concentrator) [inline wallarm/resource_rule_overlimit_res_settings.go]: Oops!

File: wallarm/resource_rule_set_response_header.go [hunk 4]
```diff
@@ -355,3 +360,78 @@ func resourceWallarmSetResponseHeaderUpdate(d *schema.ResourceData, m interface{
...
+
+		d.Set("mode", (*actionHints.Body)[0].Mode)
+		d.Set("name", (*actionHints.Body)[0].Name)
+		valuesInterface := (*actionHints.Body)[0].Values
+		var valuesStr []string
+		for _, item := range valuesInterface {
+			str, _ := item.(string)
+			valuesStr = append(valuesStr, str)
+		}
+		d.Set("values", valuesStr)
```
Comment (concentrator) [inline wallarm/resource_rule_set_response_header.go]: Here you set `name` and `values`, but in create and read functions this is used: 
```
headers := d.Get("headers").(map[string]interface{})
```
Terraform resource has different representation of headers than API:
```
  headers = {
    Server = "Wallarm solution"
    Blocked = "Blocked by Wallarm"
  }
```
VS
```
{
    "action": [
        ...
    ], 
    "mode": "append", 
    "name": "Server", 
    "values": [
        "Wallarm solution"
    ], 
}
```

File: wallarm/resource_rule_vpatch.go [hunk 5]
```diff
@@ -471,6 +471,11 @@ func resourceWallarmVpatchImport(d *schema.ResourceData, m interface{}) ([]*sche
 			}
 		}
 
+		d.Set("attack_type", (*actionHints.Body)[0].AttackType)
```
Comment (concentrator) [inline wallarm/resource_rule_vpatch.go:449]: Something is wrong here. In terraform `vpatch` rule accepts array of attack types, but in `disable_attack_type` rule it's a single string value!
```
resource "wallarm_rule_disable_attack_type" "disable_sqli" {
  action {
    type = "iequal"
    value = "example.com"
    point = {
      header = "HOST"
    }
  }
  point = [["get_all"]]
  attack_type = "sqli"
}
```
VS
```
resource "wallarm_rule_vpatch" "default" {
  attack_type =  ["sqli"]
  point = [["get", "query"]]
}
```
I think someone was trying to make a composite terraform resource to support multiple `vpatch` rules with different attack types in a single resource, which is opposite to the concept of a separate rule for each attack type in API.

File: wallarm/resource_rule_attack_rechecker_rewrite.go [hunk 6]
```diff
@@ -349,3 +354,70 @@ func resourceWallarmAttackRecheckerRewriteUpdate(d *schema.ResourceData, m inter
...
+					return nil, err
+				}
+				actsSlice = append(actsSlice, acts)
+				actionsSet.Add(acts)
+			}
+			if err := d.Set("action", &actionsSet); err != nil {
+				return nil, err
+			}
+		}
+
```
Comment (concentrator) [inline wallarm/resource_rule_attack_rechecker_rewrite.go:412]: Looks like `point` is forgotten. And in resource Schema both `point` and `rules` should have `ForceNew: true`  (we can't update the rule in API i guess).

File: wallarm/resource_rule_overlimit_res_settings.go (hunk 1 above)
Comment (KNechaevWallarm) [reply wallarm/resource_rule_overlimit_res_settings.go:178]: copy-past

**Pull request merged**